OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/MyCoursesListAdapter.kt

```diff
@@ -0,0 +1,114 @@
+package org.edx.mobile.view.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import org.edx.mobile.R
+import org.edx.mobile.core.IEdxEnvironment
+import org.edx.mobile.databinding.RowCourseListBinding
+import org.edx.mobile.extenstion.setVisibility
+import org.edx.mobile.model.api.CourseEntry
+import org.edx.mobile.model.api.EnrolledCoursesResponse
+import org.edx.mobile.util.images.CourseCardUtils
+import org.edx.mobile.util.images.ImageUtils
+import org.edx.mobile.util.images.TopAnchorFillWidthTransformation
+
+private val itemDiffCallback = object : DiffUtil.ItemCallback<EnrolledCoursesResponse>() {
+    override fun areItemsTheSame(
+        oldItem: EnrolledCoursesResponse,
+        newItem: EnrolledCoursesResponse,
+    ): Boolean {
+        return oldItem.courseId == newItem.courseId
+    }
+
+    override fun areContentsTheSame(
+        oldItem: EnrolledCoursesResponse,
+        newItem: EnrolledCoursesResponse,
+    ): Boolean {
+        return oldItem == newItem
+    }
+}
+
+abstract class MyCoursesListAdapter(
```

Comment:
IMO, it doesn't need a doc string (also because I don't know to write). 

ListAdapter is preferred for dynamic content because it computes differences between Lists on a background thread.